torch/utils/cpp_extension.py

```diff
@@ -291,16 +291,24 @@ def _get_sycl_arch_list():
 # If arch list returned by _get_sycl_arch_list() is empty, then sycl kernels will be compiled
 # for default spir64 target and avoid device specific compilations entirely. Further, kernels
 # will be JIT compiled at runtime.
-def _get_sycl_target_flags():
+def _append_sycl_targets_if_missing(cflags):
+    if any(flag.startswith('-fsycl-targets=') for flag in cflags):
+        # do nothing: user has manually specified sycl targets
+        return
     if _get_sycl_arch_list() != '':
-        return ['-fsycl-targets=spir64_gen,spir64']
-    return ['']
+        cflags.append('-fsycl-targets=spir64_gen,spir64')
 
-def _get_sycl_device_flags():
+def _get_sycl_device_flags(cflags):
+    try:
+        # we need last occurence of -fsycl-targets as it will be the one taking effect,
+        # so searching in reversed list
+        flags = [next(f for f in reversed(cflags) if f.startswith('-fsycl-targets='))]
+    except StopIteration:
```

Comment:
`_append_sycl_targets_if_missing` will ensure that `-fsycl-targets=` must be inside cflags, right?
So we should assert here or remove this catch except.

Reply:
> _append_sycl_targets_if_missing will ensure that -fsycl-targets= must be inside cflags, right?

Actually, in the initial version you've reviewed, this was not so. On the empty arch list, `-fsycl-targets=` was not being added at all, execution relied on the default target (spir64). But I think that's not transparent, so I've updated the code and now `_append...` always adds the flag (if not added explicitly by user).

The code here also got simplified dropping try...catch.